src/onyx/datomic/client.clj

```diff
@@ -0,0 +1,101 @@
+(ns onyx.datomic.client
+  (:require [clojure.string :as str]
+            [datomic.client.api :as d]
+            [onyx.datomic.protocols :as dp]
+            [taoensso.timbre :as log])
+  (:import [java.net URI]))
+
+(def tempid-counters (atom -1000000))
+
+(defn- _cloud-client [{:keys [datomic-cloud/system
+                              datomic-cloud/region
+                              datomic-cloud/query-group
+                              datomic-cloud/endpoint
+                              datomic-cloud/proxy-port] :as datomic-config}]
+  (let [region (or region (System/getenv "DATOMIC_CLOUD_REGION"))
+        system (or system (System/getenv "DATOMIC_CLOUD_SYSTEM"))
+        proxy-port (or proxy-port (System/getenv "DATOMIC_CLOUD_PROXY_PORT"))
+        proxy-port (when (string? proxy-port) (Integer/parseInt proxy-port))]
+    (d/client  {:server-type :cloud
+                :region region
+                :system system
+                :query-group (or query-group (System/getenv "DATOMIC_CLOUD_QUERY_GROUP") system)
+                :endpoint (or endpoint (format "http://entry.%s.%s.datomic.net:8182/" system region))
```

Comment:
Is hardcoding for `8182` intentional?

Reply:
Yes, the port number seems not configurable, and if necessary, user can override the entire endpoint.